qolmat/analysis/holes_characterization.py

```diff
@@ -88,7 +101,545 @@ def test(self, df: pd.DataFrame) -> float:
             diff_means = obs_mean - ml_means[list(tup_pattern)]
             inv_sigma_pattern = np.linalg.inv(ml_cov[:, tup_pattern][tup_pattern, :])
```

Comment:
suggestion : use np.linalg.solve for more stable computation of the inverse covariance matrix:
inv_sigma_pattern = np.linalg.solve(ml_cov[:, tup_pattern][tup_pattern, :], np.eye(len(tup_pattern)))